pruner/pruner.go

```diff
@@ -0,0 +1,339 @@
+package pruner
+
+import (
+	"sync"
+	"time"
+
+	quamina "quamina/lib"
+)
+
+// Stats reports basic counts to aid in deciding when to Rebuild.
+type Stats struct {
+	// Some of these values are ints instead of uints because Go
+	// likes to print uints in hex, and I'd like to see some
+	// temporary logging output in decimal.  ToDo: back to uints?
```

Comment:
I'd stay with ints.  Probably means you have to write an is-positive check in your API

Reply:
I'll try to add some gear to unit tests to verify that no count goes negative.

Reply:
Hmm, I'm wondering if `Emitted` and `Filtered` should still be `int64` so overflow isn't feasible on any platform.

Reply:
Oh, good catch.  Definitely.  Somebody is guaranteed to leave one of these things running for weeks.

Reply:
where by weeks I mean years.